### PR TITLE
Add S3 user guide section for Minio

### PR DESCRIPTION
**Pull request by dimas-b**

### CODE REVIEW

File: site/docs/guides/spark-s3.md [hunk 1]
```diff
@@ -54,7 +54,52 @@ for the most up-to-date information on how to connect Iceberg to S3.
 Note: the word `nessie` in configuration property names is the name of the Nessie catalog in the Spark session.
 A different name can be chosen according the user's liking.
 
-Then, in `spark-sql` issue a `use` statement to make `nessie` the current catalog:
+## Setting up Spark Session for Minio
+
+Start a local Minio server using Docker:
```
Comment (snazy) [inline site/docs/guides/spark-s3.md:59]: Maybe add a first sentence, that the only major difference is the `spark.sql.catalog.<catalog-name>.s3.endpoint` config.

File: site/docs/guides/spark-s3.md [hunk 2]
```diff
@@ -54,7 +54,52 @@ for the most up-to-date information on how to connect Iceberg to S3.
...
+
+```shell
+docker run -p 9000:9000 -p 9001:9001 --name minio \
+ -e "MINIO_ROOT_USER=datauser" -e "MINIO_ROOT_PASSWORD=minioSecret" \
+ quay.io/minio/minio:latest server /data --console-address ":9001"
+```
+
+Configure an AWS profile (e.g. called `demo`) in `~/.aws/credentials` (or other location appropriate for your OS) and
```
Comment (snazy) [inline site/docs/guides/spark-s3.md:67]: Wonder if the example should use `AWS_ACCESS_KEY_ID` + `AWS_SECRET_ACCESS_KEY` environment variables for simplicity over the profile approach (same for the AWS S§ walkthrough above).

Could add a link to https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html if people are interested how keys are looked up.

File: site/docs/guides/spark-s3.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ docker run -p 19120:19120 ghcr.io/projectnessie/nessie:latest
 Note: this example will run the Nessie Server using in-memory storage for table metadata. If/when the container
 is deleted, Nessie data about table changes will be lost, yet the data files in S3 will remain.
 
-## Setting up Spark Session
+## Setting up Spark Session for Amazon S3
 
 Configure an AWS profile (e.g. called `demo`) in `~/.aws/credentials` (or other location appropriate for your OS)
```
Comment (adutra) [inline site/docs/guides/spark-s3.md:21]: 1. Missing period
2. Maybe link users to https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html ? It wasn't obvious for me how to create that file.

File: site/docs/guides/spark-s3.md (hunk 2 above)
Comment (adutra) [reply site/docs/guides/spark-s3.md:67]: +1 for presenting 2 different methods, and also +1 for linking users to official docs (the link @snazy gave is even better than mine).

File: site/docs/guides/spark-s3.md [hunk 4]
```diff
@@ -54,7 +54,52 @@ for the most up-to-date information on how to connect Iceberg to S3.
...
+
+```shell
+docker run -p 9000:9000 -p 9001:9001 --name minio \
+ -e "MINIO_ROOT_USER=datauser" -e "MINIO_ROOT_PASSWORD=minioSecret" \
+ quay.io/minio/minio:latest server /data --console-address ":9001"
+```
+
+Configure an AWS profile (e.g. called `demo`) in `~/.aws/credentials` (or other location appropriate for your OS) and 
+put relevant Minio credentials in that profile. In this example the credentials are: `aws_access_key_id = datauser`, 
+`aws_secret_access_key = minioSecret`. Then,export the profile name in the `AWS_PROFILE` environment variable.
```
Comment (adutra) [inline site/docs/guides/spark-s3.md:69]:
```suggestion
`aws_secret_access_key = minioSecret`. Then, export the profile name in the `AWS_PROFILE` environment variable.
```

File: site/docs/guides/spark-s3.md (hunk 2 above)
Comment (dimas-b) [reply site/docs/guides/spark-s3.md:67]: TBH, I'm hesitant to use env. vars containing passwords in guides like this as I think that practice is not ideal from the security POV. I much prefer having credentials in a local file, which I think offers stronger protection.

File: site/docs/guides/spark-s3.md (hunk 3 above)
Comment (dimas-b) [reply site/docs/guides/spark-s3.md]: rewrote this paragraph

File: site/docs/guides/spark-s3.md (hunk 4 above)
Comment (dimas-b) [reply site/docs/guides/spark-s3.md]: rephrased

**Review by adutra — APPROVED** [review]

**Pull request merged**